plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/visitors/LocalScope.java

```diff
@@ -500,7 +500,47 @@ public void setScopeEndLine(int beginLine) {
                 }
                 continue;
             }
-            if (!(entry.node instanceof Assert)) {
+            if (entry.node instanceof Assert) {
+                Assert assrt = (Assert) entry.node;
+                if (assrt.test instanceof Call) {
+                    Call call = (Call) assrt.test;
+                    String rep = NodeUtils.getFullRepresentationString(call.func);
+                    if (rep == null) {
+                        continue;
+                    }
+                    Integer classIndex = ISINSTANCE_POSSIBILITIES.get(FullRepIterable.getLastPart(rep).toLowerCase());
+                    if (classIndex != null) {
+                        if (call.args != null && (call.args.length >= Math.max(classIndex, 1))) {
+                            //in all cases, the instance is the 1st parameter.
+                            String foundActTok = NodeUtils.getFullRepresentationString(call.args[0]);
+
+                            if (foundActTok != null && foundActTok.equals(actTok)) {
+                                if (classIndex > 0) {
+                                    exprType type = call.args[classIndex - 1];
+
+                                    if (type instanceof Tuple) {
+                                        //case: isinstance(obj, (Class1,Class2))
+                                        Tuple tuple = (Tuple) type;
+                                        for (exprType expr : tuple.elts) {
+                                            addRepresentationIfPossible(ret, expr);
+                                        }
+                                    } else {
+                                        //case: isinstance(obj, Class)
+                                        addRepresentationIfPossible(ret, type);
+                                    }
+                                } else {
+                                    //zope case Interface.implementedBy(obj) -> Interface added
+                                    ret.add(new TypeInfo(FullRepIterable.getWithoutLastPart(rep)));
+                                }
+                            }
+                        }
+                    }
+                }
+            } else if (entry.node instanceof Call && isTypingCast((Call) entry.node)) {
+                Call call = (Call) entry.node;
+                exprType type = call.args[0];
```

Comment:
It just occurred to me that you must also check if `call.args != null && call.args.length > 0`.